src/Elmish/Component.purs

```diff
@@ -53,12 +55,18 @@ import Elmish.Trace (traceTime)
 -- |
 -- |     update :: State -> Message -> Transition Aff Message State
 -- |     update state (ChildMsg m) =
--- |         bimap ChildMsg (state { child = _ }) $ Child.update state.child m
+-- |         Child.update state.child m
+-- |         # bimap ChildMsg (state { child = _ })
 -- |
-data Transition m msg state = Transition state (Array (m msg))
+data Transition m msg state = Transition state (Array (Command m msg))
+
+-- | An effect that is launched as a result of a component state transition.
+-- | It's a function that takes a callback that allows it to produce (aka
+-- | "dispatch") messages.
+type Command m msg = (msg -> Effect Unit) -> m Unit
 
 instance trBifunctor :: Functor m => Bifunctor (Transition m) where
-    bimap f g (Transition s cmds) = Transition (g s) (f <$$> cmds)
+    bimap f g (Transition s cmds) = Transition (g s) (cmds <#> \cmd sink -> cmd $ sink <<< f)
```

Comment:
Where does the name `sink` come from?

Reply:
In engineering "sink" is a traditional name for a device that takes stuff away somewhere. Compare with "heat sink" or "money sink". In software this term becomes specialized to a mechanism that takes events or messages and sends them somewhere, we don't care where. See https://en.m.wikipedia.org/wiki/Sink_(computing)

Reply:
> Even your kitchen sink kinda does that if you sink about it.

🤦